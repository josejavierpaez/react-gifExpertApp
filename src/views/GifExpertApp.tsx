import React, { useState } from 'react'
import { AddCategory } from '../components/AddCategory'
import { GifGrid } from '../components/GifGrid/GifGrid';

export const GifExpertApp = ({ defaulCategories = ['One Punch'] }) => {



  const [categories, setcategories] = useState(defaulCategories)


  // const handleAdd = (newCategory: string) => {
  //   setcategories(cats => [...cats, newCategory]);
  // }
  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setcategories} />
      <hr />

      <ol>
        {
          categories.map(category => {
            return (
              <GifGrid
                key={category}
                category={category}
              />
            )
          })
        }
      </ol>
    </>
  );
}

