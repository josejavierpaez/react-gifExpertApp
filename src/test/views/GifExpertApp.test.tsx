import { shallow } from 'enzyme';
import { GifExpertApp } from '../../views/GifExpertApp'

describe('test to <GifExpertApp/> component', () => {
  test('should be render <GifExpertApp/> component well', () => {
    const categories = ['One Punch', 'goku'];
    const wrapper = shallow(<GifExpertApp defaulCategories={categories} />);

    expect(wrapper).toMatchSnapshot()

    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })

})