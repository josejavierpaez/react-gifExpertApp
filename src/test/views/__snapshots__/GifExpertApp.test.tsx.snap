// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to <GifExpertApp/> component should be render <GifExpertApp/> component well 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="goku"
      key="goku"
    />
  </ol>
</Fragment>
`;
